src/QueueReceiver.Core/Services/PersonService.cs

```diff
@@ -75,17 +79,25 @@ public async Task UnVoidPersonAsync(long personId)
 
             if (person?.Oid != null && await _graphService.AdPersonFoundInDeletedDirectory(person.Oid))
             {
-                return await CreatePerson(adPerson, shouldReconcile: true);
+                person.Oid = userOid;
+                return;
             }
 
             if (person != null)
             {
                 person.Oid = userOid;
-                return person;
+                return;
+            }
```

Comment:
What is the difference in these two if's ?
It looks to me like they do more or less the same, or the result is anyways: OID is set on person if it exist and we end with return.
Leftovers from old logic? Is it enough with the last one?

Reply:
You're right. I've commented out the top one for  now as I'm not 100% sure what we should do in this scenario. drodling needed.